Bottom: f36ac99c2b81360c21d3a88ffbf8d6ee9369d66e
Top:    faf4f18530583aa764148a24cbc80dfb881ae3c7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-22 21:04:17 +0000




---

diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index 90977e2..4390210 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -3679,7 +3679,7 @@ ghid_ui_info_append (const gchar * new)
   if (new_ui_info_sz == 0) 
     {
       new_ui_info_sz = 1024;
-      new_ui_info = (gchar *) calloc ( new_ui_info_sz, sizeof (gchar));
+      new_ui_info = (gchar *) leaky_calloc ( new_ui_info_sz, sizeof (gchar));
     }
 
   while (strlen (new_ui_info) + strlen (new) + 1 > new_ui_info_sz)
@@ -3688,7 +3688,7 @@ ghid_ui_info_append (const gchar * new)
       gchar * np;
 
       n = new_ui_info_sz + 1024;
-      if ( (np = realloc (new_ui_info, n)) == NULL)
+      if ( (np = leaky_realloc (new_ui_info, n)) == NULL)
 	{
 	  fprintf (stderr, "ghid_ui_info_append():  realloc of size %ld failed\n",
 		   (long int) n);
